tiledb/cf/creator.py

```diff
@@ -0,0 +1,1141 @@
+              dense TileDB array (false).
+
+        Parameters:
+            array_name: Name of the array to set properties for.
+            properties: Dictionary of properties to set.
+        """
+        array_creator = self._array_creators[array_name]
+        for property_name, value in properties.items():
+            setattr(array_creator, property_name, value)
+
+    def set_attr_properties(self, attr_name: str, properties: Dict[str, Any]):
+        """Sets properties for an attribute in the CF dataspace.
+
+        Valid properties are:
+            * ``name``: The name of the attribute.
+            * ``dtype``: Numpy dtype of the attribute.
+            * ``fill``: Fill value for unset cells.
+            * ``var``: Specifies if the attribute is variable length (automatic for
+              bytes/strings).
+            * ``nullable``: Specifies if the attribute is nullable using validity tiles.
+            * ``filters``: Specifies compression filters for the attributes.
+
+        Parameters:
+            attr_name: Name of the attribute to set properties for.
+            properties: Dictionary of properties to set.
+        """
+        if "name" in properties:
+            old_name = attr_name
+            attr_name = properties.pop("name")
+            self.rename_attr(old_name, attr_name)
+        try:
+            array_name = self._attr_to_array[attr_name]
+        except KeyError as err:
+            raise KeyError(
+                f"Attribute with name '{attr_name}' does not exist."
+            ) from err
+        self._array_creators[array_name].set_attr_properties(
+            attr_name,
+            properties,
+        )
+
+    def to_schema(self, ctx: Optional[tiledb.Ctx] = None) -> GroupSchema:
+        """Returns a group schema for the CF dataspace.
+
+        Parameters:
+           ctx: If not ``None``, TileDB context wrapper for a TileDB storage manager.
+        """
+        array_schemas = {
+            array_name: array_creator.to_schema(ctx)
+            for array_name, array_creator in self._array_creators.items()
+        }
+        group_schema = GroupSchema(array_schemas)
+        group_schema.set_default_metadata_schema(ctx)
+        return group_schema
+
+
+class DataspaceConverter(DataspaceCreator):
+    @classmethod
+    def from_netcdf(cls, *args, **kwargs):
+        """Returns a DataspaceConverter with a NetCDF4ConvertEngine generated from the
+        provided NetCDF file.
+
+        See :class:`~tiledb.cf.engine.netcdf4_engine.NetCDF4ConverterEngine` for more
+        information.
+
+        Parameters:
+            args: Positional arguments to pass to NetCDF4ConverterEngine.from_file
+            kwargs: Keyword arugments to pass to NetCDF4ConverterEngine.from_file
+        """
+        from .engines.netcdf4_engine import NetCDF4ConverterEngine
+
+        return cls(NetCDF4ConverterEngine.from_file(*args, **kwargs))
+
+    def __init__(self, converter_engine):
```

Comment:
My plan is to write a protocol for the converter engines as well as some clear documentation on how to create a converter class. That way someone can create a standalone converter engine that does not need to subclass anything from this library. This would be in a future pull request.